app/electron/runCmd.ts

```diff
@@ -173,9 +185,27 @@ export function handleRunCommand(
     return;
   }
 
-  const child: ChildProcessWithoutNullStreams = spawn(commandData.command, commandData.args, {
+  const [permissionsValid, permissionError] = checkPermissionSecret(commandData, permissionSecrets);
+  if (!permissionsValid) {
+    console.error(permissionError);
+    return;
+  }
+
+  // Get the command and args to run. With the correct paths for "scriptjs" commands.
+  // scriptjs commands are scripts run with the compiled app, or with "Electron" in dev mode.
+  const command = commandData.command === 'scriptjs' ? process.execPath : commandData.command;
+  const args =
+    commandData.command === 'scriptjs'
+      ? [getPluginsScriptPath(commandData.args[0]), ...commandData.args.slice(1)]
+      : commandData.args;
+
+  const child: ChildProcessWithoutNullStreams = spawn(command, args, {
     ...commandData.options,
     shell: false,
+    env: {
+      ...process.env,
+      ...(commandData.command === 'scriptjs' ? { HEADLAMP_RUN_SCRIPT: 'true' } : {}),
```

Comment:
So this works even if we do not have the RUN_SCRIPT as true? Maybe leave a comment explaining why this behavior.

Reply:
I added some comments explaining that this passes the HEADLAMP_RUN_SCRIPT env var, and why it does that.

  // If the command is 'scriptjs', we pass the HEADLAMP_RUN_SCRIPT=true 
  // env var so that the Headlamp or Electron process runs the script.